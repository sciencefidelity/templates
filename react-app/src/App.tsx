import { useState } from "react"

export function App() {
  const [count, countSet] = useState(0)
  const title = "Welcome to Vite.js!"
  return (
    <main className="container">
      <div>
        <h1>{title}</h1>
        <button onClick={() => countSet(count + 1)}>Count is: {count}</button>
      </div>
    </main>
  )
}
