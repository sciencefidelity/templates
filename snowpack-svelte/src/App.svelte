<script lang="ts">
  import { Router, Route } from "svelte-routing"
  import About from "@/routes/About.svelte"
  import Counter from "@/routes/Counter.svelte"
  import Footer from "@/lib/Footer.svelte"
  import Header from "@/lib/Header.svelte"
  import Home from "@/routes/Home.svelte"
  import Navbar from "@/lib/Navbar.svelte"
  import NotFound from "@/routes/404.svelte"
</script>

<div id="app">
  <Router>
    <div>
      <Header />
      <Navbar />
    </div>
    <main>
      <Route path="/"><Home /></Route>
      <Route path="about" component={About} />
      <Route path="counter" component={Counter} />
      <Route component={NotFound} />
    </main>
    <Footer />
  </Router>
</div>

<style lang="scss">
  #app {
    display: grid;
    grid-template-rows: auto 1fr auto;
    place-items: center;
    height: 100vh;
    background-color: var(--background);
  }
</style>
