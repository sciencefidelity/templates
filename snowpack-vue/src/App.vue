<script lang="ts">
import Vue from 'vue'
import Component from 'vue-class-component'

// Define the component in class-style
@Component
export default class Counter extends Vue {
  // Class properties will be component data
  count = 0

  // Methods will be component methods
  increment() {
    this.count++
  }

  decrement() {
    this.count--
  }
}
</script>

<template>
  <header class="header">
    <img src="/logo.svg" class="header__logo" alt="logo" />
    <p>
      Page has been open for <code>{{ count }}</code> seconds.
    </p>
    <a
      class="App-link"
      href="https://vuejs.org"
      target="_blank"
      rel="noopener noreferrer"
    >
      Learn Vue
    </a>
  </header>
</template>

<style lang="scss">
@keyframes logo-pulse {
  from {
    transform: scale(1);
  }
  to {
    transform: scale(1.06);
  }
}

.header {
  background-color: #f9f6f6;
  color: #333;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  &__logo {
    height: 36vmin;
    pointer-events: none;
    margin-bottom: 3rem;
    animation: logo-pulse infinite 1.6s ease-in-out alternate;
  }
}
</style>
