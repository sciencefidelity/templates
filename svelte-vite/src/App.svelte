<script lang="ts">
  import svelteLogo from './assets/svelte.svg?url'
  import Counter from './lib/Counter.svelte'
</script>

<main>
  <div>
    <a href="https://vitejs.dev" target="_blank">
      <img src="/vite.svg" class="logo" alt="Vite Logo" />
    </a>
    <a href="https://svelte.dev" target="_blank">
      <img src={svelteLogo} class="logo svelte" alt="Svelte Logo" />
    </a>
  </div>
  <h1>Vite + Svelte</h1>

  <div class="card">
    <Counter />
  </div>

  <p class="read-the-docs">
    Click on the Vite and Svelte logos to learn more
  </p>
</main>

<style>
  .logo {
    height: 6em;
    padding: 1.5em;
    will-change: filter;
  }
  .logo:hover {
    filter: drop-shadow(0 0 2em hsla(237, 100%, 70%, 0.667));
  }
  .logo.svelte:hover {
    filter: drop-shadow(0 0 2em hsla(15, 100%, 50%, 0.667));
  }
  .read-the-docs {
    color: hsl(0, 0%, 53%);
  }
</style>
