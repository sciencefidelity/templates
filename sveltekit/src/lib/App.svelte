<script lang="ts">
	import '../app.css';
	import Counter from '$lib/Counter.svelte';
	import svelteLogo from '../assets/svelte.svg?url';
</script>

<main>
	<div>
		<a href="https://kit.svelte.dev" target="_blank">
			<img src={svelteLogo} class="logo" alt="Svelte Logo" />
		</a>
	</div>
	<h1>Svelte Kit</h1>

	<div class="card">
		<Counter />
	</div>

	<p class="read-the-docs">Click on the Svelte logo to learn more</p>
</main>

<style>
	.logo {
		height: 6em;
		padding: 1.5em;
		will-change: filter;
	}

	.logo:hover {
		filter: drop-shadow(0 0 2em hsla(15, 100%, 50%, 0.667));
	}
	.read-the-docs {
		color: hsl(0, 0%, 53%);
	}
</style>
