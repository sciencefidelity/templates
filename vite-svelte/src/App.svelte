<script>
  import logo from "./assets/logo.svg"
</script>

<main>
  <img src={logo} alt="Svelte Logo" />
  <h1>Hello world!</h1>
</main>

<style>

</style>
