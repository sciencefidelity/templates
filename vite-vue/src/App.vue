<script lang="ts">import { Vue, Options } from "vue-class-component"
import Footer from "./components/Footer.vue"
import Header from "./components/Header.vue"
// import Layout from "./layouts/Layout.vue"
import Navbar from "./components/Navbar.vue"

@Options({
  name: "App",
  components: {
    Footer,
    Header,
    // Layout,
    Navbar
  }
})
export default class App extends Vue {

}
</script>

<template>
  <div id="app">
    <div>
      <Header />
      <Navbar />
    </div>
    <main>
<!--       <Layout> -->
        <router-view />
<!--       </Layout> -->
    </main>
    <Footer />
  </div>
</template>

<style lang="scss">
@use './styles/foundation/colors';

header,
footer {
  color: var(--dark-green);
  padding: 2rem 0;
  text-align: center;
  width: 100vw;
}

footer {
  border-top: 1px solid var(--dark-green);
}

nav {
  background-color: var(--dark-green);
  color: var(--background);
  display: grid;
  font-weight: 600;
  padding: 1rem 2rem;
  place-items: center;
  text-align: center;
  width: 100%;
  a {
    color: var(--background);
    text-decoration: none;
    &:hover {
      text-decoration: underline;
    }
  }
  ul {
    display: flex;
    list-style: none;
    margin: 0;
    padding: 0;
    li {
      padding: 0 2rem;
    }
  }
}

#app {
  display: grid;
  grid-template-rows: auto 1fr auto;
  height: 100vh;
  place-items: center;
}
</style>
