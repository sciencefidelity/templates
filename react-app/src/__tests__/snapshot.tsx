import { render } from "@testing-library/react"
import { App } from "../App"

it("renders homepage unchanged", () => {
  const { container } = render(<App />)
  expect(container).toMatchSnapshot()
})
