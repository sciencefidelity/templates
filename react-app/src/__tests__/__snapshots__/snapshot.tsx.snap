// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <main
    class="container"
  >
    <div>
      <h1>
        Welcome to Vite.js!
      </h1>
      <button>
        Count is: 
        0
      </button>
    </div>
  </main>
</div>
`;
